core/src/main/java/com/yahoo/ycsb/Client.java

```diff
@@ -801,67 +820,76 @@ public void run()
       System.exit(0);
     }
 
-    try
-    {
-      workload.init(props);
-    }
-    catch (WorkloadException e)
-    {
-      e.printStackTrace();
-      e.printStackTrace(System.out);
-      System.exit(0);
-    }
+    final Tracer tracer = new Tracer.Builder("YCSB " + workload.getClass().getSimpleName()).conf(conf).build();
+
+    try (final TraceScope span = tracer.newScope("Client#workload_init")) {
+
+      try
+      {
+        workload.init(props);
+      }
+      catch (WorkloadException e)
+      {
+        e.printStackTrace();
+        e.printStackTrace(System.out);
```

Comment:
HTrace registers a shutdown hook with java.lang.Runtime#addShutdownHook.  This hook closes all SpanReceiver objects when the JVM exits.  SpanReceiver#close will attempt to persist the spans to htraced (or wherever that particular SpanReceiver sends spans).  In the case of HTracedSpanReceiver it will try to send them to htraced.

This shutdown hook will be run in the System.exit(0) case you guys are talking about.  It is also run in the SIGTERM (i.e. user sends control-C or normal kill) case.  It's not run in SIGKILL, but that is by necessity.  SIGKILL _always_ kills the process immediately without giving it the chance to do any cleanup.

Bottom line: this is not an issue.  You don't need a finally block here, or to worry about flushing anything on shutdown.